pandas/core/arrays/period.py

```diff
@@ -52,13 +71,17 @@ def _period_array_cmp(cls, op):
 
     def wrapper(self, other):
         op = getattr(self._ndarray_values, opname)
+        if isinstance(other, (ABCSeries, ABCIndexClass)):
+            # TODO: return NotImplemented?
```

Comment:
Yes, return NotImplemented.  Also presumably ABCDataFrame.

Reply:
This will fail some tests because we reverse the error message.

```
arrays/period.py:86: AssertionError: "Input has different freq=A-DEC from PeriodIndex" does not match "Input has different freq=M from PeriodIndex(freq=A-DEC)"
```

I think that's OK though.